lib/lints/use_common_design_system_widget/node_handler/widget_node_handler.dart

```diff
@@ -0,0 +1,91 @@
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:chili_custom_lints/lints/use_common_design_system_widget/widget_helper/widget_helper.dart';
+import 'package:collection/collection.dart';
+
+typedef DirectionalSpacing = (double? horizontal, double? vertical);
+
+class WidgetNodeHandler {
+  static String getCorrectionName(
+    String prefix,
+    double? value,
+    DesignSystemElement element,
+  ) =>
+      '$prefix${element.name}${value?.toInt()}';
+
+  static String getCorrectionMessage(String correctionName) =>
+      "Use common design system widget '$correctionName' instead";
+
+  static bool hasLintError(
+    double? horizontal,
+    double? vertical,
+    NodeList<Expression> arguments,
+  ) {
+    final isHorizontalDivisibleBy4 = horizontal != null && horizontal % 4 == 0;
+    final isVerticalDivisibleBy4 = vertical != null && vertical % 4 == 0;
+    final hasArguments = arguments.isNotEmpty;
+
+    return isHorizontalDivisibleBy4 || isVerticalDivisibleBy4 || !hasArguments;
+  }
+
+  static DirectionalSpacing getMarginArgumentValues(
+    InstanceCreationExpression node,
+  ) {
+    final width = _getArgumentValue(
+      node: node,
+      parameterName: SizedBoxParameter.width.name,
+      valueStartPosition: SizedBoxParameter.width.valueStartPosition,
+    );
+
+    final height = _getArgumentValue(
+      node: node,
+      parameterName: SizedBoxParameter.height.name,
+      valueStartPosition: SizedBoxParameter.height.valueStartPosition,
+    );
+
+    return (width, height);
+  }
+
+  static DirectionalSpacing getPaddingArgumentValues(
+    InstanceCreationExpression node,
+  ) {
+    final horizontal = _getArgumentValue(
+      node: node,
+      parameterName: EdgeInsetsParameter.horizontal.name,
+      valueStartPosition: EdgeInsetsParameter.horizontal.valueStartPosition,
+    );
+    final vertical = _getArgumentValue(
+      node: node,
+      parameterName: EdgeInsetsParameter.vertical.name,
+      valueStartPosition: EdgeInsetsParameter.vertical.valueStartPosition,
+    );
+
+    return (horizontal, vertical);
+  }
+
+  static double? _getArgumentValue({
+    required InstanceCreationExpression node,
+    required String parameterName,
+    required int valueStartPosition,
+  }) {
+    // For handling unnamed parameter value.
+    final allArguments = node.argumentList.arguments.toSet();
+    final unnamedValue = double.tryParse(allArguments.firstOrNull.toString());
+
+    if (allArguments.length == 1 && unnamedValue != null) {
+      return unnamedValue;
+    }
+
+    final argument = node.argumentList.arguments.toSet().firstWhereOrNull(
```

Comment:
You already have this `node.argumentList.arguments.toSet()` in the `allArguments` variable.